compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1002,6 +1006,9 @@ pub struct TargetOptions {
     /// Is true if the target is an ARM architecture using thumb v1 which allows for
     /// thumb and arm interworking.
     pub has_thumb_interworking: bool,
+
+    /// Whether the target supports the wasi reactor execution model
+    pub supports_wasi_reactor: bool,
```

Comment:
Adding a new target option is probably an overkill right now, you could check `target.os == "wasi"` instead.

Reply:
Alright, wasn't sure if a string comparison would be too hacky